src/Repository/HealthCheckConfigRepository.php

```diff
@@ -0,0 +1,172 @@
+<?php
+
+namespace PrestaShop\Module\HealthCheck\Repository;
+
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\Driver\Statement;
+use Doctrine\DBAL\FetchMode;
+use Doctrine\DBAL\Query\QueryBuilder;
+use PrestaShop\Module\HealthCheck\Entity\HealthCheckConfig;
+use PrestaShop\PrestaShop\Core\Exception\DatabaseException;
+
+class HealthCheckConfigRepository
+{
+    /**
+     * @var Connection
+     */
+    private $connection;
+
+    /**
+     * @var string
+     */
+    private $tableName;
+
+    public function __construct(
+        Connection $connection,
+        string $dbPrefix
+    ) {
+        $this->connection = $connection;
+        $this->tableName = sprintf('%shealth_check_config', $dbPrefix);
+    }
+
+    /**
+     * @return HealthCheckConfig|null
+     */
+    public function getLastHealthCheckConfig()
+    {
+        /** @var QueryBuilder $qb */
+        $qb = $this->connection->createQueryBuilder();
+        $qb
+            ->select('id_health_check_config as id, token, ips')
```

Comment:
```suggestion
            ->select('MAX(id_health_check_config) AS id, token, ips')
```

And remove the where

Reply:
If the user insert 2 rows in ps_health_check_config
if you use MAX(), you need to have GROUP BY and when you do so, you will have the 2 rows

Reply:
ERROR 1140 (42000): In aggregated query without GROUP BY, expression #2 of SELECT list contains nonaggregated column 'prestashop.ps_health_check_config.token'; this is incompatible with sql_mode=only_full_group_by

Reply:
Ok. Thanks for the explanation.